### ISSUE TITLE
Nan value error

### ISSUE THREAD

**Issue by mrpahdosw** [ultralytics/yolov5#11280]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

The values of box_loss obj_loss and cls_loss appear **nan** when running the model on gpu but shows loss if ran on cpu.

![image](https://user-images.githubusercontent.com/129479877/229023546-a73c8ea1-4125-4237-b805-440302d73d1f.png)

### Environment

-YOLO - YOLOv5 (torch==1.12.1+cu113 torchvision==0.13.1+cu113 torchaudio==0.12.1)
-CUDA - 11.3
-OS - Windows 11

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by tasawor** [comment]
@mrpahdosw did you fix it?

**Comment by glenn-jocher** [comment]
@tasawor, I apologize for any confusion but I am not aware of a solution for your question. It appears that the loss is `nan` when running on the GPU but is working fine when running on the CPU. Could you please provide more information regarding your dataset, training script, gradient accumulation steps, batch size, and GPU model so that I can better understand the issue and help you to fix it?

In the meantime, you can try inspecting the input `torch.Tensor` values passed to the criterion function and verify they are not `nan`. For further troubleshooting, you can utilize PyTorch's built-in `autograd` debugging utilities, such as `autograd.detect_nan` as in this documentation - https://pytorch.org/docs/stable/notes/autograd.html#debugging-nan-issues.

**Comment by mrpahdosw** [comment]
@glenn-jocher I have tried on multiple dataset but the issue was the same as well as for different batch sizes, training script was train.py with no changes and GPU is GTX 1650.

**Comment by mrpahdosw** [comment]
@tasawor I have tried with python version 3.7 and it is working but not resolved with python 3.10

**Comment by glenn-jocher** [comment]
@mrpahdosw, thank you for providing more information. 

It's good to hear that the issue is resolved in Python 3.7. Keep in mind that PyTorch and its dependencies can be sensitive to Python version and CUDA version, so it's always a good idea to verify your environment is set up properly. 

If you find that the issue persists while using Python 3.10, I would recommend trying to run the training script in Python 3.7 within a virtual environment or container to help isolate the issue.

Additionally, make sure that you have installed the latest version of PyTorch compatible with Python 3.10 and CUDA 11.3. You can also try updating your CUDA toolkit by following NVIDIA's instructions for your operating system.

**Comment by mrpahdosw** [comment]
@glenn-jocher I have been doing training in virtual environment with the all the  possible compatible version of PyTorch with Python 3.10 and CUDA 11.3 but the issue is still there, so I have worked with Python 3.7 only.

**Comment by glenn-jocher** [comment]
@mrpahdosw, I'm sorry to hear that the issue persisted even with the latest compatible versions of PyTorch with Python 3.10 and CUDA 11.3. However, it's great to know that you were able to resolve the issue with Python 3.7. 

If you encounter any other issues in the future, feel free to reach out and we'll be happy to help you out. And, don't forget to check out the YOLOv5 documentation on our website, https://docs.ultralytics.com/yolov5, for further guidance on training and using the YOLOv5 model.
